tests/experimental/test_experimental_base_simulator.py

```diff
@@ -0,0 +1,198 @@
+
+    # Check shape
+    assert output.shape == (1, len(mock_simulator._output_names))
+
+    # Check values against expected transformation
+    expected_sum = 0.5 + 0.0 + 7.5
+    assert output[0, 0] == pytest.approx(expected_sum * 1)
+    assert output[0, 1] == pytest.approx(expected_sum * 2)
+
+
+def test_forward_batch(mock_simulator):
+    """Test that forward_batch method processes multiple samples correctly"""
+    # Create test batch
+    n_samples = 3
+    batch = torch.tensor(
+        [[0.5, 0.0, 7.5], [0.2, 0.3, 6.0], [0.8, -0.5, 9.0]], dtype=torch.float32
+    )
+
+    # Process batch
+    results = mock_simulator.forward_batch(batch)
+
+    # Check shape
+    assert results.shape == (n_samples, len(mock_simulator._output_names))
+
+    # Check values for each sample
+    for i in range(n_samples):
+        expected_sum = sum(batch[i].tolist())
+        assert results[i, 0] == pytest.approx(expected_sum * 1)
+        assert results[i, 1] == pytest.approx(expected_sum * 2)
+
+
+def test_get_parameter_idx(mock_simulator):
+    """Test that get_parameter_idx returns correct indices"""
+    assert mock_simulator.get_parameter_idx("param1") == 0
+    assert mock_simulator.get_parameter_idx("param2") == 1
+    assert mock_simulator.get_parameter_idx("param3") == 2
+
+    # Test invalid parameter
+    with pytest.raises(ValueError, match="Parameter .* not found"):
+        mock_simulator.get_parameter_idx("invalid_param")
+
+
+def test_properties(mock_simulator, parameters_range, output_names):
+    """Test that properties return correct values"""
+    # Test param_names
+    assert mock_simulator.param_names == list(parameters_range.keys())
+
+    # Test param_bounds
+    assert mock_simulator.param_bounds == list(parameters_range.values())
+
+    # Test output_names
+    assert mock_simulator.output_names == ["var1", "var2"]
+
+    # Test output_names
+    assert mock_simulator.output_names == output_names
+
+
+def test_abstract_class():
+    """Test that Simulator cannot be instantiated directly"""
+    params = {"param": (0, 1)}
+    outputs = ["out"]
+
+    # Should raise TypeError when trying to instantiate abstract base class
+    with pytest.raises(TypeError):
+        Simulator(params, outputs)  # type: ignore[abstract]
+
+
+def test_handle_simulation_failure():
+    """Test handling of simulation failures in forward_batch"""
+
+    # This test demonstrates that the current implementation doesn't properly handle
+    # None returns (which would be a good enhancement)
+    class ThresholdSimulator(MockSimulator):
+        def _forward(self, x: TensorLike) -> TensorLike:
+            # Only process inputs where the first value is > 0.5
+            if x[0, 0] > 0.5:
+                return super()._forward(x)
+            # For other cases, let's return a tensor of zeros instead of None
+            # since the type system doesn't allow None returns
+            return torch.zeros((x.shape[0], len(self._output_names)))
```

Comment:
Perhaps we can add:
```suggestion
            # For other cases, let's return a tensor of zeros instead of None
            # since the type system doesn't allow None returns
            # TODO (#438): update to handle failed simulations
            return torch.zeros((x.shape[0], len(self._output_names)))
```